server/server.ts

```diff
@@ -0,0 +1,191 @@
+import bcrypt from 'bcryptjs'
+import { Request, Response, NextFunction } from 'express'
+import jsonServer from 'json-server'
+import jwt from 'jsonwebtoken'
+import { dirname } from 'path'
+import { fileURLToPath } from 'url'
+
+import { DbUser, convertDbUserToUser } from './dbModels'
+import { User, Post, Gender } from '../src/models'
+
+// Type for the database schema
+interface Database {
+  users: DbUser[]
+  posts: Post[]
+}
+
+// Get the directory name of the current module file (for db.json path)
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = dirname(__filename)
+
+// Create a new jsonServer instance
+const server = jsonServer.create()
+
+// Middleware to parse JSON bodies
+server.use(jsonServer.bodyParser)
+
+// Assuming `router.db` is a Lowdb instance with a JSON backend
+const router = jsonServer.router<Database>(`${__dirname}/db.json`)
+// Use default middlewares (for CORS, logging, etc.)
+const middlewares = jsonServer.defaults()
+
+interface RegisterRequestBody {
+  firstName: string
+  lastName: string
+  email: string
+  password: string
+  confirmPassword: string
+  gender: Gender
+}
+
+// Register a custom POST /register endpoint
+server.post('/register', (req: Request, res: Response) => {
+  const {
+    firstName,
+    lastName,
+    email,
+    password,
+    confirmPassword,
+    gender,
+  }: RegisterRequestBody = req.body
+
+  if (password !== confirmPassword) {
+    res
+      .status(400)
+      .json({ message: 'Password and Confirm Password should match' })
+    return
+  }
+
+  if (!firstName || !lastName || !email || !password || !gender) {
+    res.status(400).json({
+      message: 'Firstname, lastname, email, gender, and password are required',
+    })
+    return
+  }
+
+  // Check if user already exists
+  const dbUsers = router.db.get('users').value()
+  const userExists = dbUsers.find((dbUser) => dbUser.email === email)
```

Comment:
You have to check for existing user 2 times, in the register and login form, therefor, would be great to have a helper function called **findUserByEmail**, that would receive email and the users (the db).